tests/integration/use-test.js

```diff
@@ -45,4 +48,76 @@ module('@use', () => {
 
     assert.equal(instance.test, 'world');
   });
+
+  module('async updates', function() {
+    test('it works', async function(assert) {
+      class Invoker extends Resource {
+        @tracked result = Infinity;
+
+        get value() {
+          return this.result;
+        }
+
+        @action
+        @waitFor
+        async invoke() {
+          let { fn, args } = this.args.named;
+
+          await Promise.resolve();
+
+          this.result = await fn(...args);
+        }
+
+        setup() {
+          this.invoke();
+        }
+
+        update() {
+          this.invoke();
+        }
+      }
+
+      class MyClass {
+        @tracked left = 0;
+        @tracked right = 0;
+
+        @use test = new Invoker(() => {
+          return {
+            named: {
+              fn: this.math,
+              args: [this.left, this.right],
+            }
+          }
+        });
+
+        @action
+        @waitFor
+        math(left, right) {
+          return Promise.resolve(left + right);
+        }
+      }
+
+      let instance = new MyClass();
+
+      assert.equal(instance.test, Infinity, 'initial value');
+      await settled();
+
+      assert.equal(instance.test, 0, 'after async functions run');
+
+      instance.left = 1;
+      await settled();
+
+      assert.equal(instance.test, 1);
```

Comment:
This assertion fails as `instance.test` is `0`.
I don't think it's timing related, because `await this.pauseTest()` doesn't change anything.